app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -345,6 +347,13 @@ class SettingsFragment : PreferenceFragmentCompat() {
         preferenceMakeDefaultBrowser.onPreferenceClickListener =
             getClickListenerForMakeDefaultBrowser()
 
+        preferenceOpenLinksInExternalApp?.setOnPreferenceChangeListener<Boolean> { preference, newValue ->
+            preference.context.settings().preferences.edit()
+                .putBoolean(preference.key, newValue).apply()
+            requireComponents.core.engine.settings.remoteDebuggingEnabled = newValue
```

Comment:
We probably don't want to set this engine setting right? Can we just use `SharedPreferenceUpdater()` as the `onPreferenceChangeListener` if we don't need side effects?

Reply:
Sorry, this was supposed to be a draft :)